projects/http-loader/src/lib/http-loader.ts

```diff
@@ -39,9 +48,43 @@ export class TranslateHttpLoader implements TranslateLoader {
     public getTranslation(lang: string): Observable<TranslationObject> {
         const cacheBuster = this.config.enforceLoading ? `?enforceLoading=${Date.now()}` : "";
 
-        return this.http.get(
-            `${this.config.prefix}${lang}${this.config.suffix}${cacheBuster}`,
-        ) as Observable<TranslationObject>;
+        if (this.config.ressources.length > 0) {
+            const requests = this.config.ressources.map((resource) => {
+                let path: string;
+
+                if (typeof resource === "string") path = `${resource}${lang}.json`;
+                else path = `${resource.prefix}${lang}${resource.suffix ?? ".json"}`;
+
+                return this.http.get<TranslationObject>(`${path}${cacheBuster}`).pipe(
+                    catchError((err) => {
+                        if (
+                            this.config.showLog ||
+                            (resource as TranslateHttpLoaderResource).showLog
+                        ) {
+                            console.error(`Error loading translation for ${lang}:`, err);
+                        }
+                        return of({});
+                    }),
+                );
+            });
+
+            return forkJoin(requests).pipe(
+                map((response) => response.reduce((acc, curr) => mergeDeep(acc, curr), {})),
+            ) as Observable<TranslationObject>;
+        }
+
+        return this.http
```

Comment:
This is almost a duplication of the code above. I think we could handle this by always using the resources array instead.
If a single file is added, we can create resources array with 1 entry instead. 
Allowing both is only confusing. Who is going to win? obviously, when providing the resources array, the other one is ignored. but this is absolutely not transparent for the user. the resources loader also does not implement the cache-buster.


we could also use provideHttpLoader() and proviceMultiHttpLoader() or similar to make it clear what is happening here.

Reply:
@CodeAndWeb

1.  I would indeed only use the resource as this is a cleaner solution, but this would add a breaking changes. so I didn't wanted it '^^.

2. I'll refactor it right away

3. the resource loader does implement the cache-buster 
```typescript
return this.http.get<TranslationObject>(`${path}${cacheBuster}`).pipe(
```

4. I'm against adding a new provider, as it only complexify what could be done with a single one.